File: ios/RCTCameraManager.m
#import <Foundation/Foundation.h>
#import <React/RCTViewManager.h>
#import <React/RCTBridgeModule.h>

// RCT_EXTERN_MODULEの第二引数にはこのクラスの型
// Viewを伴う場合RCTViewManager, ネイティブ機能のみの場合NSObject
@interface RCT_EXTERN_MODULE(RCTCamaraManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(flash, BOOL)

RCT_EXTERN_METHOD(capture:(nonnull NSNumber *)node)
@end

File: ios/RCTCameraManager.swift
import UIKit
import AVFoundation
import Foundation

@objc(RCTCamaraManager)
class RCTCamaraManager: RCTViewManager {
  override func view() -> UIView! {
    // 他のUIVewでラップしてリターンした方がいい場合もある。
    // https://reactnative.dev/docs/native-components-ios.html#ios-mapview-example
    return CameraView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func capture(_ node: NSNumber) {
    let component = getCameraView(tag: node)
    component.capture()
  }
  
  func getCameraView(tag: NSNumber) -> CameraView {
    return bridge.uiManager.view(forReactTag: tag) as! CameraView
  }
}
